[docs] Reword palette intention and fix format (#21477)
mui-org/material-ui: docs/src/pages/components/popover/popover.md

## Mouse over interaction

> This demonstrates how to use the `Popover` component to implement a popover behavior based on the mouseover event.

Reviewer: I know this was previously used but 'to implement a popover behaviour' doesn't sound right to me. 

I think something like this sounds much better:
```suggestion
This demonstrates how to use the `Popover` component and the mouseover event to achieve popover behavior.
```

Author: Yep definitely makes more sense ~
